### PR TITLE
Fix/matomo hard-coded url and csp

### PR DESCRIPTION
**Pull request by bertrandmc**
# Description

- Replace hard-coded matomo and configure CSP rules. 
- Enforce secure urls for matomo, pay api and pay return. 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have manually tested the application

# Checklist:

- [X] My changes do not introduce any new linting errors
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation and versioning
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have rebased onto master and there are no code conflicts
- [X] I have checked deployments are working in all environments
- [X] I have updated the architecture diagrams as per Contribute.md

### CODE REVIEW

**Comment by bertrandmc** [comment]
Relates to https://github.com/XGovFormBuilder/digital-form-builder/pull/238

File: runner/src/server/views/layout.html [hunk 1]
```diff
@@ -34,7 +34,7 @@
       _paq.push(['trackPageView']);
       _paq.push(['enableLinkTracking']);
       (function() {
-        var u="https://web-analytics.fco.gov.uk/piwik/";
+        var u="{{ matomoUrl }}/piwik/";
```
Comment (trevorrea) [inline runner/src/server/views/layout.html:37]: HI @bertrandmc - you need to change the noscript line too to be 

    <noscript><p></p></noscript>

Comment (bertrandmc) [reply runner/src/server/views/layout.html:37]: Updated. Thanks for the head-up @trevorrea.

Comment (bertrandmc) [reply runner/src/server/views/layout.html:37]: Thanks for the head-up @trevorrea

**Review by jenbutongit — APPROVED** [review]

**Pull request merged**